cli/cli_app.go

```diff
@@ -277,15 +330,25 @@ func processHooks(hooks []config.Hook, terragruntOptions *options.TerragruntOpti
 		allPreviousErrors := append(previousExecError, errorsOccurred...)
 		if shouldRunHook(curHook, terragruntOptions, allPreviousErrors...) {
 			terragruntOptions.Logger.Printf("Executing hook: %s", curHook.Name)
-			actionToExecute := curHook.Execute[0]
-			actionParams := curHook.Execute[1:]
-			possibleError := shell.RunShellCommand(terragruntOptions, actionToExecute, actionParams...)
-
-			if possibleError != nil {
-				terragruntOptions.Logger.Printf("Error running hook %s with message: %s", curHook.Name, possibleError.Error())
-				errorsOccurred = append(errorsOccurred, possibleError)
+			oldEnvMap, err := loadEnvVars(terragruntOptions, curHook)
+			if nil != err {
```

Comment:
`err != nil`

Reply:
Normally I prefer to have the constant always on the left side so that the compiler will create an error when I miss the second '=' in an equality comparison and make it up into an assignment.

https://en.wikipedia.org/wiki/Yoda_conditions

Reply:
I'm familiar with yoda conditions but they aren't necessary in Go. The following code won't even compile:

```go
if err = nil {
  // ...
}
```

And since yoda conditions are a bit weird to read, no need for them here.